### ISSUE TITLE
Error in Laravel 5.6.37

### ISSUE THREAD

**Issue by dmitrijsmihailovs** [czim/laravel-paperclip#26]
When I updated Laravel to 5.6.37, paperclip started to throw this error - Object of class Czim\Paperclip\Attachment\Attachment could not be converted to string
On 5.6.35 it works fine.

**Comment by kirba** [comment]
This is happening because Laravel changed performInsert method in Illuminate\Database\Eloquent\Model.php
Now in peformInsert method `$attributes = $this->getAttributes();` is used to get attributes instead of `$attributes = $this->attributes`

The quick fix would be to override getAttributes in your model where you are using PaperclipTrait
```php
public function getAttributes() 
{
    return parent::getAttributes();
}
```

**Comment by czim** [comment]
Good catch. I've resolved this, temporarily, in `2.5.3`. A better solution is still required, but I can't think of one right off the bat, so I've opted to go with something of a work-around/hack.

`debug_backtrace()` is used to detect wether `performInsert()` is called and prevents the (uninsertable) attachments from being injected. The only clean way to fix this that I can think of, is a way to differentiate between 'in-class' versus external calls to `getAttributes()` -- which is exactly the problem that directly accessing the `attributes` property obviated.

Any ideas welcome, of course!
